src/main/core/choo-choo-train.ts

```diff
@@ -60,69 +60,126 @@ interface LogPayload {
+
+import * as fs from 'fs'
+
+const LOCK_FILE = path.join(os.tmpdir(), 'ibmtelemetry.lock')
+const WC_LOCK_FILE = path.join(os.tmpdir(), 'ibmtelemetry-wc.lock')
+
+const IPC_ADDR = path.join(os.tmpdir(), 'ibmtelemetry-ipc')
+const WC_IPC_ADDR = path.join(os.tmpdir(), 'ibmtelemetry-wc-ipc')
+
 /**
  * Encapsulates all logic for orchestrating the running of multiple telemetry processes.
  */
 export class ChooChooTrain extends Loggable {
   private readonly workQueue: Work[] = []
-  private readonly ipcAddr: string
+  private ipcAddr: string
   private analyzedCommit?: string
   private analyzedPath?: string
+  private configPath: string
   private date?: string
   private environment?: Environment
   private gitInfo?: GitInfo
+  private parsedConfig?: Record<string, unknown> & ConfigSchema
   private projectId?: string
   private scanId?: string
   private logEndpoint?: string
   private totalDuration?: number
   private totalPackages?: number
+  private isConductor: boolean
 
   /**
    * Constructs a new ChooChooTrain instance.
    *
-   * @param ipcAddr - The address of the IPC pipe.
    * @param environment - Environment variable configuration for this run.
    * @param configFilePath - Path to a config file.
    * @param logger - A logger instance.
    */
-  public constructor(
-    ipcAddr: string,
-    environment: Environment,
-    configFilePath: string,
-    logger: Logger
-  ) {
+  public constructor(environment: Environment, configFilePath: string, logger: Logger) {
     super(logger)
 
-    this.ipcAddr = ipcAddr
+    this.isConductor = false
+    this.ipcAddr = IPC_ADDR
+
+    this.logger.debug('Working environment:', environment.cwd)
+    this.configPath = configFilePath
 
     this.workQueue.push({ cwd: environment.cwd, configFilePath })
   }
 
   /**
    * Establishes ourself as either the conductor or a client.
+   * Attempts connection to an existing conductor 3 times.
+   * If not connection is made, it attempts to become a client instead.
+   *
    * If we are the conductor, run all work in the queue (including our work).
    * If we are a client, send our work to the conductor.
    */
   public async run(): Promise<void> {
     let connection: net.Socket | net.Server | undefined
+    let lockFileName: string = LOCK_FILE
+
+    this.parsedConfig = await this.getPackageData(this.configPath)
+
+    // is now using the Web Components server
+    const wc = this.parsedConfig?.collect?.wc
+
+    if (wc !== undefined) {
+      this.logger.debug('Package has web components scope')
+      this.ipcAddr = WC_IPC_ADDR
+      lockFileName = WC_LOCK_FILE
+    }
+
+    // Try to acquire the lock
+    try {
+      const fd = fs.openSync(lockFileName, 'wx') // atomic creation
+      fs.writeSync(fd, `${process.pid}`)
+      fs.closeSync(fd)
+      this.isConductor = true
 
-    for (let i = 0; i < MAX_RETRIES && !connection; i++) {
-      // Try to be the server
+      this.logger.debug('This is now a web components IPC server')
```

Comment:
```suggestion
this.logger.debug('This is a new IPC server')
```
this will also get logged for a non-web components IPC server